core/src/agent/state.rs

```diff
@@ -133,15 +194,19 @@ fn reduce_commit_entry(
     let chain_header = create_new_chain_header(&entry, state);
 
     fn response(
+        context: Arc<Context>,
         state: &mut AgentState,
         entry: &Entry,
         chain_header: &ChainHeader,
     ) -> Result<Address, HolochainError> {
         state.chain.content_storage().add(entry)?;
         state.chain.content_storage().add(chain_header)?;
+        let agent = AgentStateSnapshot::new(chain_header.clone());
+        state.chain.content_storage().add(&agent)?;
```

Comment:
Why adding the agent snapshot to the CAS?
And how could we get it out again, if that is meant as the persistence of the agent state?

Reply:
at first i was surprised/concerned by this

now i think it's actually important to do it this way, any given `AgentStateSnapshot` only makes sense for a specific CAS - at the hodl podl we said that we wanted to keep the "many CAS" option cleanly available

we can hardcode the `.address()` in the `AddressableContent` implementation for `AgentStateSnapshot` to something predictable and that should sidestep a whole bunch of potential "dangling snapshot" issues where a snapshot points to a missing/out of sync CAS.

Reply:
Hm, a deterministic address for the snapshot was the missing puzzle piece, at least for my perspective. Yeah, that sounds doable and interesting :)